src/pages/index.js

```diff
@@ -43,82 +45,193 @@ const imageModal = document.querySelector("#image__modal");
-    },
-  },
-  "#cards__list"
-);
-
-section.renderItems();
+// Create image popup instance
+const imagePopup = new PopupWithImage("#image__modal");
+imagePopup.setEventListeners();
 
-profileEditButton.addEventListener("click", () => {
-  const userData = userInfo.getUserInfo();
-  profileFormPopup.setInputValues({
-    name: userData.name,
-    description: userData.job,
+// Fetch user info and initial cards
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      job: userData.about,
+      avatar: userData.avatar,
+      id: userData._id,
+    });
+
+    const section = new Section(
+      {
+        items: cards,
+        renderer: (item) => {
+          const cardElement = createCard(item);
+          section.addItem(cardElement);
+        },
+      },
+      "#cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.error("Error loading app data:", err);
   });
-  profileFormPopup.open();
-});
 
-addProfileButton.addEventListener("click", () => {
-  addCardPopup.open();
+const deleteCardPopup = new PopupWithConfirmation("#delete__card-modal", () => {
+  api
+    .deleteCard(cardToDeleteId)
+    .then(() => {
+      cardToDeleteElement.remove();
+      deleteCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    })
+    .finally(() => {
+      deleteCardPopup.setButtonText("Yes");
+    });
+  console.log(cardToDeleteId);
 });
 
-// Universal form validators
-const formValidators = {};
+deleteCardPopup.setEventListeners();
 
-const enableValidation = (config) => {
-  const formList = Array.from(document.querySelectorAll(config.formSelector));
-  formList.forEach((formElement) => {
-    const validator = new FormValidator(config, formElement);
-    const formName = formElement.getAttribute("name");
-    formValidators[formName] = validator;
-    validator.enableValidation();
-  });
-};
+let cardToDeleteId = null;
+let cardToDeleteElement = null;
 
-enableValidation(FormValidatorObjects);
+function handleDeleteCard(cardId, cardElement) {
+  cardToDeleteId = cardId;
+  cardToDeleteElement = cardElement;
+  deleteCardPopup.open();
+}
+
+function handleFormSubmit(popupInstance, apiMethod, successCallback) {
+  popupInstance.setButtonText("Saving...");
+  apiMethod
+    .then((response) => {
+      successCallback(response);
+      popupInstance.close();
+    })
+    .catch((err) => {
+      console.error("Error:", err);
+    })
+    .finally(() => {
+      popupInstance.setButtonText("Save");
+    });
+}
 
+// Profile Form Submission
+const profileFormPopup = new PopupWithForm("#profile__modal", (data) => {
+  handleFormSubmit(
+    profileFormPopup,
+    api.updateProfile({ name: data.name, about: data.description }),
+    (userData) => {
+      profileName.textContent = userData.name;
+      profileDescription.textContent = userData.about;
```

Comment:
Only methods of the `UserInfo` class should set/get the profile data (including the **avatar**)